packages/prettier-plugin-liquid/src/printer/printer-liquid-html.ts

```diff
@@ -228,17 +228,30 @@ function printNode(
       };
 
       if (isRawMarkupIdentationSensitive()) {
-        return node.value;
+        return [node.value.trimEnd(), hardline];
       }
 
+      const parentNode = node.parentNode;
+      const shouldNotIndentBody =
+        parentNode &&
+        parentNode.type === NodeTypes.LiquidRawTag &&
+        parentNode.name === 'schema' &&
+        !options.indentSchema;
+      const shouldIndentBody = node.kind !== RawMarkupKinds.markdown && !shouldNotIndentBody;
       const lines = bodyLines(node.value);
-
       const rawFirstLineIsntIndented = !!node.value.split(/\r?\n/)[0]?.match(/\S/);
       const shouldSkipFirstLine = rawFirstLineIsntIndented;
 
-      return lines.length > 0 && lines.find((line) => line.trim() !== '')
-        ? join(hardline, reindent(lines, shouldSkipFirstLine))
-        : softline;
+      const body =
+        lines.length > 0 && lines.find((line) => line.trim() !== '')
+          ? join(hardline, reindent(lines, shouldSkipFirstLine))
+          : softline;
+
+      if (shouldIndentBody) {
+        return [indent([hardline, body]), hardline];
+      } else {
+        return [dedentToRoot([hardline, body]), hardline];
```

Comment:
Had to duplicate this logic slightly because it's complicated to explain 😓. 

Basically the parent node would not know if we were able to embed or not (i.e. switch rendering from this plugin's to JS, or TypeScript, or whatever). But we need to know **if it failed** in order to remove the hardlines and print as is without the surrounding hardlines (which we do on line 231).